plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php

```diff
@@ -623,12 +627,36 @@ public static function get_all_statuses() {
 			);
 			/* phpcs:enable */
 
-			Cache::set( $cache_key, $statuses );
+			set_transient( 'woocommerce_analytics_orders_statuses_all', $statuses, YEAR_IN_SECONDS );
 		}
 
 		return $statuses;
 	}
 
+	/**
+	 * Ensure the order status will present in `get_all_statuses` call result.
+	 *
+	 * @param int $order_id Order ID.
+	 */
+	public static function actualize_all_statuses_transient_if_necessary( $order_id ) {
```

Comment:
Nitpicky, but maybe the following to keep similar semantics across the repo:

```suggestion
	public static function maybe_update_order_statuses_transient( $order_id ) {
```

Reply:
Addressed in fee108fe4ce27baf12ebb39997a297865d196944